lib/pbench/agent/results.py

```diff
@@ -144,58 +132,66 @@ def make_md5sum(self, tarball):
                     self.logger.error(
                         "error removing failed tar ball MD5, {}", tarball_md5
                     )
-            sys.exit(1)
+            raise RuntimeError("Error in making tarball.md5", exc)
 
 
 class CopyResultTb:
     chunk_size = 4096
 
     def __init__(self, controller, tarball, config, logger):
         self.tarball = Path(tarball)
-        if not self.tarball.exists():
-            logger.error("tarball does not exist, '{}'", tarball)
-            sys.exit(1)
         self.tarball_md5 = Path(f"{tarball}.md5")
-        if not self.tarball_md5.exists():
-            logger.error("tarball's .md5 does not exist, '{}'", self.tarball_md5)
-            sys.exit(1)
         self.logger = logger
-        server_rest_url = config.results.get("server_rest_url")
-        self.upload_url = f"{server_rest_url}/upload/ctrl/{controller}"
+        server_rest_url = config.get("results", "server_rest_url")
+        self.upload_url = f"{server_rest_url}/upload/{controller}/localhost"
+
+    def check_tb_tbmd5(self):
+        if not self.tarball.exists():
+            raise FileNotFoundError("tarball does not exist, '{}'", self.tarball)
+        if not self.tarball_md5.exists():
+            raise FileNotFoundError(
+                "tarball's .md5 does not exist, '{}'", self.tarball_md5
+            )
 
     def read_in_chunks(self, file_object):
         data = file_object.read(self.chunk_size)
         while data:
             yield data
             data = file_object.read(self.chunk_size)
 
-    def copy_result_tb(self):
+    def copy_result_tb(self, token):
         files = [f for f in self.tarball.parent.iterdir() if f.is_file()]
         file_count = len(files)
         if file_count != 2:
-            self.logger.error(
+            raise Exception(
                 "(internal): unexpected file count, {}, associated with tarball, '{}'",
                 file_count,
                 self.tarball,
             )
-            sys.exit(1)
 
         with open(self.tarball_md5, "r") as md5fp:
             md5sum = md5fp.read().split()[0]
         filename = secure_filename(str(self.tarball))
-        headers = {"filename": filename, "Content-MD5": md5sum}
+        headers = {
+            "filename": filename,
+            "Content-MD5": md5sum,
+            "Authorization": f"Bearer {token}",
+        }
         with self.tarball.open("rb") as f:
             try:
                 response = requests.put(
                     self.upload_url, data=self.read_in_chunks(f), headers=headers
                 )
                 self.logger.info("File uploaded successfully")
-            except Exception:
-                self.logger.exception("There was something wrong with your request")
-                sys.exit(1)
-        if not response.status_code == 200:
-            self.logger.error(
+            except FileUploadError as exc:
```

Comment:
Shouldn't this be `Exception`?

Reply:
yeah.. Fixed

Reply:
And `raise FileUploadError(...) from exc` so we have traceabilility.

Reply:
sure